services/src/main/java/org/keycloak/services/resources/admin/UsersResource.java

```diff
@@ -107,7 +109,20 @@ public UsersResource(RealmModel realm, AdminPermissionEvaluator auth, AdminEvent
     @POST
     @Consumes(MediaType.APPLICATION_JSON)
     public Response createUser(final UserRepresentation rep) {
-        auth.users().requireManage();
```

Comment:
I did not catch this before, but I think we should first try to check `requireManage` and only in case we get an exception (lacking permission) we fallback to your changes that rely on the fine-grained permission.

The reason behind this comment is that we avoid people not using fine-grained permissions to not be impacted by the additional checks you are adding.